.github/workflows/newdeploy.yml

```
@@ -0,0 +1,352 @@
+  
+  build-go-binary:
+    needs: [ EnvSetup , build-scan-publish ]
+    runs-on: self-hosted
+    name: Build Go Binary
+    steps:
+
+      - name: Checkout code
+        uses: actions/checkout@0ad4b8fadaa221de15dcec353f45205ec38ea70b # v4.1.4
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
```

actionlint (1.7.4)

the runner of "actions/setup-go@v3" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)